dotCMS/src/main/java/com/dotcms/rest/api/v1/folder/FolderHelper.java

```diff
@@ -98,4 +100,56 @@ public Folder loadFolderByURI(String siteName, User user, String uri) throws Dot
         return ret;
     }
 
+    /**
+     *
+     * @param hostId hostId where the folder lives
+     * @param folder parent folder to find
+     * @param user user making the request
+     * @return FolderView with the info of the folder requested and the subFolders
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    public FolderView loadFolderAndSubFoldersByPath(final String hostId, final String folder, final User user)
+            throws DotSecurityException, DotDataException {
+        final String uriParam = !folder.startsWith(StringPool.FORWARD_SLASH) ? StringPool.FORWARD_SLASH.concat(folder) : folder;
+        final Host host = APILocator.getHostAPI().find(hostId,user,false);
+        final Folder folderByPath = APILocator.getFolderAPI().findFolderByPath(uriParam, host, user, false);
```

Comment:
Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5916479)